src/ocl/convolutionocl.cpp

```diff
@@ -245,12 +248,61 @@ EvaluateDataImplicitGemmSolution(Handle& handle,
             elapsed += handle.GetKernelTime();
         }
         else if(kernel.GetName() ==
-                "gridwise_convolution_backward_data_implicit_gemm_v4r1_nchw_kcyx_nkhw")
+                    "gridwise_convolution_backward_data_implicit_gemm_v1r1_nchw_kcyx_nkhw" ||
+                kernel.GetName() ==
+                    "gridwise_convolution_backward_data_implicit_gemm_v1r1_ncdhw_kczyx_nkdhw")
+        { // only pad=0 stride=1 filter 1 not set zero
+            bool is_1x1_s1 = false;
+            if(miopen::all_of(padding, [](auto v) { return v == 0; }) &&
+               miopen::all_of(strides, [](auto v) { return v == 1; }))
+            {
+                if(wDesc.GetLengths()[2] == 1 && wDesc.GetLengths()[3] == 1)
+                { // filter = 1
+                    if(wDesc.GetSize() == 4 || (wDesc.GetSize() == 5 && wDesc.GetLengths()[4] == 1))
+                    {
+                        is_1x1_s1 = true;
+                    }
+                }
+            }
+
+            if(!is_1x1_s1)
+            {
+                float zero = 0.f;
+                SetTensor(handle, outDesc, out, &zero);
+                elapsed += handle.GetKernelTime();
+            }
+        }
+        else if(kernel.GetName() ==
+                    "gridwise_convolution_backward_data_implicit_gemm_v4r1_nchw_kcyx_nkhw" ||
+                kernel.GetName() ==
+                    "gridwise_convolution_backward_data_implicit_gemm_v4r1_ncdhw_kczyx_nkdhw")
         {
-            // \todo this kernel doesn't always need to set-zero
-            float zero = 0.f;
-            SetTensor(handle, outDesc, out, &zero);
-            elapsed += handle.GetKernelTime();
+            bool is_1x1_s1 = false;
```

Comment:
[NOT URGENT] same here, please fix the naming in the future

Reply:
v4r1 condition change to filterGeStride